planning/behavior_velocity_planner/invalid-lanelet-design.md

```diff
@@ -0,0 +1,47 @@
+## Invalid Lanelet
+
+### Role
+
+This module plans the velocity of the related part of the path in case there is an invalid lanelet referring to it.
+
+![invalid_lanelet_design.svg](docs%2Finvalid_lanelet%2Finvalid_lanelet_design.svg)
+
+### Activation Timing
+
+This function is activated when the lane id of the target path has an invalid lanelet label (i.e. the `invalid_lanelet` attribute is `yes`).
+
+### Module Parameters
+
+| Parameter          | Type   | Description                                          |
+| ------------------ | ------ | ---------------------------------------------------- |
+| `stop_margin`      | double | [m] margin for ego vehicle to stop before speed_bump |
+| `print_debug_info` | bool   | whether debug info will be printed or not            |
+
+### Inner-workings / Algorithms
+
+- Get invalid lanelet attribute on the path from lanelet2 map
+- The invalid lanelet state machine starts in `INIT` state
+- Get the intersection points between path and invalid lanelet polygon
+- Assign the state to `APPROACHING` toward an invalid lanelet if:
+  - the distance from front of the ego vehicle till the first interection point between the ego path and the invalid lanelet polygon is more than the `stop_margin`
```

Comment:
@ahmeddesokyebrahim Thanks for your contribution :clap: At first, could you fix these typos ?

Reply:
Thanks for your words Mr @satoshi-ota .
Resolved as well.